src/components/Dropdown/index.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useMemo } from 'react';
+import { CSSTransition } from 'react-transition-group';
+import classNames from 'classnames';
+
+import useClickOutside from '../../hooks/useClickOutside';
+
+import Input, { IInput } from '../Input';
+import Elevation from '../Elevation';
+import Button from '../Button';
+import Icon from '../Icons';
+
+import './index.scss';
+
+const TEST_ID = '@fv/dropdown';
+
+export interface IDropdown extends Omit<IInput, 'mask' | 'type' | 'sideContent'> {
+  isOpen?: boolean;
+  closeOnBlur?: boolean;
+  closeOnClickOutside?: boolean;
+  inputClassName?: 'string';
+  onToggle?: (isOpen: boolean) => void;
+  onClickOutside?: React.MouseEventHandler<HTMLButtonElement>;
+  onIconClick?: React.MouseEventHandler<HTMLButtonElement>;
+}
+
+export const DropDown: React.FC<IDropdown> = (props) => {
+  const {
+    isOpen,
+    name,
+    children,
+    className,
+    disabled,
+    onToggle,
+    onFocus,
+    onBlur,
+    onIconClick,
+    onClickOutside,
+    inputClassName,
+    closeOnBlur,
+    closeOnClickOutside,
+    ...rest
+  } = props;
+
+  const controlled = useMemo(() => isOpen !== undefined && isOpen !== null, [isOpen]);
+  const [isOpenInternal, setOpenInternal] = useState(false);
+
+  const toggleOpenInternal = (isOpen: boolean) => {
+    setOpenInternal(isOpen);
+    onToggle && onToggle(isOpen);
+  };
+
+  const handleClickOutside = (evt: any) => {
+    onClickOutside && onClickOutside(evt);
+    if (controlled || !closeOnClickOutside) {
+      return;
+    }
+    toggleOpenInternal(false);
+  };
+  const dropdownRef = useClickOutside<HTMLDivElement>(handleClickOutside);
+
+  const handleBlur = (evt: React.FocusEvent<HTMLInputElement>) => {
+    if (dropdownRef.current?.contains(evt.relatedTarget)) {
+      return;
+    }
+
+    onBlur && onBlur(evt);
+    if (controlled && !closeOnBlur) {
+      return;
+    }
+    toggleOpenInternal(false);
+  };
+
+  const handleFocus = (evt: React.FocusEvent<HTMLInputElement>) => {
+    onFocus && onFocus(evt);
+    if (controlled) {
+      return;
+    }
+    toggleOpenInternal(true);
+  };
+
+  const handleIconClick = (evt: React.MouseEvent<HTMLButtonElement>) => {
+    onIconClick && onIconClick(evt);
+    if (controlled) {
+      return;
+    }
+    toggleOpenInternal(!isOpenInternal);
+  };
+
+  const isDropdownOpen = !disabled && (controlled ? isOpen : isOpenInternal);
```

Comment:
also here:

const isDropdownOpen = !disabled && (isOpen || isOpenInternal);

Reply:
we even dont need `controlled` const

Reply:
This is also deliberate - isOpen || isOpenInteral will fall back to "isOpenInternal" if "isOpen" is falsy. The condition above is picking which prop to use depending on wether the component is controlled or not and with "disabled" acting as an override to both of them